### DISCUSSION TITLE
locally fine tune a large language model to learn a C# code base and perform code intelligence tasks

### DISCUSSION THREAD

**Discussion by ghost** [community/community#101775] [Programming Help]
I want to fine-tune a LLM locally to serve as an intelligent code reviewer to use as a tool for developers that, given natural language descriptions, identifies and highlights specific locations in the C# codebase where changes are needed. The goal is to streamline the code review process by providing developers with precise indications of where modifications should be made based on their high-level descriptions. Even though there are suitable LLMs for the task i can't figure out a way to feed my C# code base to the LLM. (a way for the LLM to read my code files ). Any suggestions?

**Comment by samus-aran** [comment]
Hi @venushki99 , Thanks for joining us for the GitHub Universe celebrations in the GitHub Community, we're glad you're here! 

We want to make sure you’re getting the best support you can now that GitHub Universe has ended so I've gone ahead and moved this to the most relevant category for you.

Thank you for helping us maintain a productive and tidy community for all our members.
